IoTCoreDefaultApp/IoTCoreDefaultApp/Views/OOBENetwork.xaml.cs

```diff
@@ -151,7 +151,7 @@ private void ConnectButton_Clicked(object sender, RoutedEventArgs e)
             {
                 await dispatcher.RunAsync(CoreDispatcherPriority.Normal, () =>
                 {
-                    CortanaHelper.LaunchCortanaToConsentPageAsyncIfNeeded();
+                    CortanaHelper.LaunchCortanaToConsentPageAsyncIfNeeded()?.Wait();
```

Comment:
The RunAsync causes this lambda to run on the UI thread, therefore the .Wait() will block the UI. Instead add the async keyword before the () => of the lambda, and change the .Wait() to an await at the beginning of the line.

Also, the "await" before the RunAsync is not necessary.